src/mlpack/bindings/python/print_pyx.cpp

```diff
@@ -120,8 +120,7 @@ void PrintPYX(const util::BindingDetails& doc,
   for (ParamIter it = parameters.begin(); it != parameters.end(); ++it)
   {
     util::ParamData& d = it->second;
-    if (d.input)
-      IO::GetSingleton().functionMap[d.tname]["PrintClassDefn"](d, NULL, NULL);
+    IO::GetSingleton().functionMap[d.tname]["PrintClassDefn"](d, NULL, NULL);
```

Comment:
If I remember right, the `if` was here originally so that if you had an input parameter and an output parameter that both needed a definition, we did not print that definition twice.  (In retrospect I should have commented that better; sorry about that.)  So maybe we need to re-add it?

Reply:
Actually, now that we are keeping different files for each method, so, one file should output the model (here the `_fit_main.cpp`) and the other file should take in the model or input (here `_predict_main.cpp`) and for each method, a separate `.pyx` file will be created so, we should have the class definition in both the files now to prevent the `LinearRegressionType is not defined` error. That is why I removed this `if` condition.

Reply:
Ohh, good point!  It might be worthwhile to be a bit more paranoid and track which `tname`s we've already called `PrintClassDefn` for, so that we don't do it twice.  Up to you if you want to do that---I can't imagine a binding where we would output more than one model of the same type, or where we would take more than one model of the same type as input.

Reply:
Yes, I also can't think of a similar scenario but it is always better to be sure, I will work on tracking that.